### ISSUE TITLE
Parse/read old dependency format, `"x,y,z"`, in addition to `["x","y","z"]`

### ISSUE THREAD

**Issue by bradyt** [kdheepak/taskwarrior-tui#379]
I think dependency as array was an option in Taskwarrior since 2015, but was made default last August 17th, of 2021.

In other words, the default was previously for Taskwarrior output to use `"depends":"x,y,z"`, but last year `"depends":["x","y","z"]` became enforced, if I understand correctly.

I understand taskwarrior-tui does not really support Taskwarrior 2.5.3, but I would be curious if it would not be hard to have taskwarrior-tui just able to read the old dependency format. Potentially any users that have not updated to 2.6.X yet can begin using taskwarrior-tui sooner. I'm not sure what other issues to anticipate. Any other issues I'm failing to anticipate about supporting Taskwarrior 2.5.3 may demotivate this issue.

Note that Taskwarrior current stable has an outstanding bug where `task sync` can corrupt the dependency field. So using Taskserver, depends, and taskwarrior-tui, is currently not an option.

Here is an example `~/.task/pending.data`, where the fourth task depends on the first three tasks:

```
[description:"foo" entry:"1609459200" modified:"1609459200" status:"pending" uuid:"00000000-0000-0000-0000-000000000000"]
[description:"foo" entry:"1609459200" modified:"1609459200" status:"pending" uuid:"11111111-1111-1111-1111-111111111111"]
[description:"foo" entry:"1609459200" modified:"1609459200" status:"pending" uuid:"22222222-2222-2222-2222-222222222222"]
[depends:"00000000-0000-0000-0000-000000000000,11111111-1111-1111-1111-111111111111,22222222-2222-2222-2222-222222222222" description:"foo" entry:"1609459200" modified:"1609459200" status:"pending" uuid:"33333333-3333-3333-3333-333333333333"]
```

And the corresponding `task export`:

```json
[
{"id":1,"description":"foo","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"00000000-0000-0000-0000-000000000000","urgency":10},
{"id":2,"description":"foo","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"11111111-1111-1111-1111-111111111111","urgency":10},
{"id":3,"description":"foo","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"22222222-2222-2222-2222-222222222222","urgency":10},
{"id":4,"depends":"00000000-0000-0000-0000-000000000000,11111111-1111-1111-1111-111111111111,22222222-2222-2222-2222-222222222222","description":"foo","entry":"20210101T000000Z","modified":"20210101T000000Z","status":"pending","uuid":"33333333-3333-3333-3333-333333333333","urgency":-3}
]
```

**Comment by kdheepak** [comment]
Thanks for the detailed issue, that is indeed annoying. 

The other issue is the handling of command line arguments related to filter has subtly changed between 2.5.x and 2.6.x. I think I have special cases them in taskwarrior-tui but I’ve never been able to test them thoroughly.

**Comment by bradyt** [comment]
I had an epiphany, that I could just put `json.depends.array=yes` in my config and taskwarrior-tui would work. But alas, that brings me back to the issue at Taskserver, which I've filed a fresh issue at https://github.com/GothenburgBitFactory/taskserver/issues/181.

Should I try `rc.json.depends.array=yes` somewhere in taskwarrior-tui source code as a patch, to further test things out?

**Comment by kdheepak** [comment]
Oh interesting. I didn’t know about that option. Yeah, if you are going to test that I’d (and the taskwarrior-tui) community would super appreciate it. I’m a tad busy for the next couple of weeks and won’t be able to do a deep dive into this.

**Comment by bradyt** [comment]
Sounds good, I'll give the following a try:

```diff
modified   src/app.rs
@@ -1607,6 +1607,7 @@ impl TaskwarriorTui {
 
         task.arg("rc.json.array=on");
         task.arg("rc.confirmation=off");
+        task.arg("rc.json.depends.array=on");
 
         if let Some(args) = shlex::split(&format!(
             r#"rc.report.{}.filter='{}'"#,
```

**Comment by kdheepak** [comment]
Any update on this? Should this be included by default?

**Comment by bradyt** [comment]
I'm sort of on limbo on things. Apparently I now have Taskwarrior installed from source at `develop` branch, as there has been progress on the `depends` issue at https://github.com/GothenburgBitFactory/taskwarrior/issues/2689. Before that issue came to light, I simply hadn't upgraded from Taskwarrior 2.5.3, then as that issue became apparent, I held further reservations on upgrading, but as that has been fixed, and once it's released, I may likely upgrade to Taskwarrior 2.6.2. That will reduce my own personal interest/motivation in seeing apps able to parse the old depends format.

When I did try taskwarrior-tui with the above patch, and Taskwarrior 2.5.3, I simply noticed the tasks weren't sorted and filtered as they are with Taskwarrior 2.6.X, which I think you pointed out above. But the patch above at least doesn't seem to break taskwarrior-tui's ability to interact with Taskwarrior 2.6.X.
